src/main/scala/com/fulcrumgenomics/bam/FindSwitchbackReads.scala

```diff
@@ -0,0 +1,395 @@
+            val gap = later.start - earlier.end - 1
+            if (abs(gap) <= maxGap) {
+              Some(TandemBasedHit(later.start - earlier.end - 1))
+            }
+            else {
+              None
+            }
+        }
+      case _ =>
+        None
+  }
+
+  /**
+    * Attempts to find the first ungapped match of the query sequence within the target sequence. "First" is either
+    * from the start of the target sequence, or if `preferLaterMatches` is true, from the end of the target sequence.
+    *
+    * @param query the query sequence to be found
+    * @param target the target sequence in which the query sequence is to be located
+    * @param maxErrorRate the maximum rate of mismatches allowed when matching the sequences
+    * @param preferLaterMatches if true prefer matches closer to the end of targetSequence, else prefer matches closer
+    *                           to the start
+    * @return a [[ReadBasedHit]] if a match is found, otherwise None.
+    */
+  def find(query: String, target: String, maxErrorRate: Double, preferLaterMatches: Boolean = false): Option[ReadBasedHit] = {
+    val maxMismatches = Math.floor(query.length * maxErrorRate)
+    val range = {
+      if (preferLaterMatches) Range.inclusive(target.length-query.length, 0, step = -1)
+      else Range.inclusive(0, target.length-query.length-1, step=1)
+    }
+
+    range.iterator.flatMap { targetOffset =>
+      var mismatches = 0
+      forloop (from=0, until=query.length) { i =>
+        if (!SequenceUtil.basesEqual(query(i).toByte, target(targetOffset+i).toByte)) mismatches += 1
+      }
+
+      if (mismatches <= maxMismatches) Some(ReadBasedHit(offset=targetOffset, length=query.length))
+      else None
+    }.find(_ => true)
+  }
+
+  /** Returns the read number, either 1 or 2. */
+  private def readNum(rec: SamRecord): Int = if (!rec.paired || rec.firstOfPair) 1 else 2
+
+  /** Returns the concatenation of the unique values per read group, or the default if there are no values. */
+  private def summarizeRg(header: SAMFileHeader, f: SAMReadGroupRecord => String, default: String = "n/a"): String = {
+    val ss = header.getReadGroups.map(f).filter(_ != null).map(_.trim).filter(_.nonEmpty).toSeq.distinct.sorted
+    if (ss.nonEmpty) ss.mkString(",") else default
+  }
+}
+
+@clp(group=ClpGroups.Personal, description=
+  """
+    |Finds reads where a template switch occurred during library construction.  Some library construction methods,
+    |notably ultra-low-input shotgun methods, are prone to template switching events that create molecules
+    |(templates, inserts) that instead of being a linear copy of a segment of genomic DNA, instead are chimeras formed
+    |by starting on one strand of DNA and then, switching to the opposite strand.  Frequently when the switch occurs
+    |there may be a small offset between where the first strand was departed and the opposite strand joined.
```

Comment:
Add this? "This offset may be due to skipping bases on the first strand after the switch point, or skipping bases on the opposite strand after the switch point."